compiler/hash-semantics/src/passes/evaluation/mod.rs

```diff
@@ -81,55 +82,65 @@ impl EvaluationPass<'_> {
     }
 }
 
-impl<'tc> AstPass for EvaluationPass<'tc> {
+impl<E: SemanticEnv> AnalysisPass for EvaluationPass<'_, E> {
+    type Env = E;
+    fn env(&self) -> &Self::Env {
+        self.env
+    }
+
+    type PassOutput = ();
+
     fn pass_interactive(
         &self,
+        source: SourceId,
         node: ast::AstNodeRef<ast::BodyBlock>,
-    ) -> crate::diagnostics::error::SemanticResult<()> {
-        let term = self.ast_info().terms().get_data_by_node(node.id()).unwrap();
+    ) -> crate::diagnostics::definitions::SemanticResult<()> {
+        let tc = TcEnvImpl::new(self.env, source);
+        let term = self.ast_info.terms().get_data_by_node(node.id()).unwrap();
 
         // Potentially dump the TIR and evaluate it depending on flags.
-        if self.settings().semantic_settings.dump_tir {
+        if self.compiler_settings().semantic_settings.dump_tir {
             dump_tir(term);
         }
 
         // Interactive mode is always evaluated.
-        let result = self.norm_ops().with_mode(NormalisationMode::Full).normalise(term.into())?;
+        let result = tc.norm_ops().with_mode(NormalisationMode::Full).normalise(term.into())?;
         stream_less_writeln!("{}", result);
 
         Ok(())
     }
 
     fn pass_module(
         &self,
+        source: SourceId,
         node: ast::AstNodeRef<ast::Module>,
-    ) -> crate::diagnostics::error::SemanticResult<()> {
-        let mod_def_id = self.ast_info().mod_defs().get_data_by_node(node.id()).unwrap();
-        let main_call_term = self.find_and_construct_main_call()?;
+    ) -> crate::diagnostics::definitions::SemanticResult<()> {
+        let tc = TcEnvImpl::new(self.env, source);
```

Comment:
I like the name `TcEnvImpl` because it is the implementation of the `TcEnv` trait from the POV of the semantic analysis. I don't see how `TcCtx` is better since context means something else.